wg-meetings/2017-03-23.md

```diff
@@ -0,0 +1,172 @@
+* \[WIP\] inspector: hint text update [node#11207](https://github.com/nodejs/node/pull/11207)
+* What will Domain be replaced with? [node#10843](https://github.com/nodejs/node/issues/10843)
+* src: separate trace_event_common from trace_event [node#10628](https://github.com/nodejs/node/pull/10628)
+* async_hooks initial implementation [node#8531](https://github.com/nodejs/node/pull/8531)
+
+* wg: nominating jkrems [diagnostics#87](https://github.com/nodejs/diagnostics/pull/87)
+* Request to join Diagnostics WG [diagnostics#86](https://github.com/nodejs/diagnostics/pull/86)
+* \[trace\] tracking issue; out of experimental [diagnostics#84](https://github.com/nodejs/diagnostics/issues/84)
+* [async_hooks] tracking issue [diagnostics#29](https://github.com/nodejs/diagnostics/issues/29)
+
+* guides: debugging getting started guide [nodejs.org#1131](https://github.com/nodejs/nodejs.org/pull/1131)
+* blog: diag wg update and --debug deprecation [nodejs.org#1156](https://github.com/nodejs/nodejs.org/pull/1156)
+
+---
+
+## Minutes
+
+### inspector: make `debug` an alias for `inspect` [node#11441](https://github.com/nodejs/node/pull/11441)
+
+* Switch the CLI debugger to V8 inspector [node#11421](https://github.com/nodejs/node/issues/11421)
+* debug: activate inspector with \_debugProcess
+[node#11431](https://github.com/nodejs/node/pull/11431)
+
+To discuss:
+
+1. Alias `node debug script.js` to `node inspect script.js`.
+2. Alias `--debug` to `--inspect`.
+3. Deprecate `vm.runInDebugContext` API.
+4. Switch SIGUSR1 to activate Inspector.
+
+Need to decide what to change now before 8.0.0.
+
+#### 1. Alias `node debug script.js` to `node inspect script.js`.
+
+Conclusion: We should alias `node debug` to `node inspect` and include a deprecation warning.
+
+#### 2. Alias `--debug` to `--inspect`.
+
+@joshgav: Does this depend on Node@8 including V8 5.7 or 5.8?
+
+@jkrems: Even if Node@8 starts with V8 5.7 we should alias `--debug` to `--inspect`. That way we won’t have to support the old interface through the lifetime of Node@8.
+
+@joshgav: Are we concerned about breaking ecosystem? Concerns about node-inspector?
+
+@ofrobots: People have moved on to other tools.
+
+@joshgav: Any reason to not alias `--debug` and instead remove entirely?
+
+@jkrems: If end goal is to encourage `--inspect` might be best to just completely remove, reduce confusion. But might be unexpected to current users.
+
+@ofrobots: Best to keep it around as an alias for a little while.
+
+@joshgav: Propose a) making it an alias to `--inspect`, b) with a runtime deprecation warning, and c) plan to remove it in a later version entirely.
+
+@jkrems: Any objections? A: No.
+
+#### 3. Deprecate `vm.runInDebugContext` API.
+
+`vm.runInDebugContext` will go away at end of 2017, so we should deprecate now in preparation for Node 10 (April 2018).
+
+@ofrobots: Are there other use cases for this? Most common use case is `var debug = vm.runInDebugContext(‘Debug’)` to access debugging methods.
```

Comment:
The V8 inspector API ought to be the mechanism to work with the debugger going forward. We are working on adding a mechanism ([PR](https://github.com/nodejs/node/pull/11431) + follow-on) for adding a mechanism to talk to the inspector API in-process from JS which should make this simpler.

I am also curious if you/others have need to run in the debug context other than getting at the `Debug` object.

/cc @hashseed, @eugeneo.